security fix

http://bugs.gentoo.org/215702
ftp://ftp.openbsd.org/pub/OpenBSD/patches/4.3/common/001_openssh.patch

Index: usr.bin/ssh/session.c
===================================================================
RCS file: /cvs/src/usr.bin/ssh/session.c,v
retrieving revision 1.230
diff -u -r1.230 session.c
--- usr.bin/ssh/session.c	22 Feb 2008 05:58:56 -0000	1.230
+++ usr.bin/ssh/session.c	27 Mar 2008 10:54:55 -0000
@@ -878,8 +878,9 @@
 	do_xauth =
 	    s->display != NULL && s->auth_proto != NULL && s->auth_data != NULL;
 
-	/* ignore _PATH_SSH_USER_RC for subsystems */
-	if (!s->is_subsystem && (stat(_PATH_SSH_USER_RC, &st) >= 0)) {
+	/* ignore _PATH_SSH_USER_RC for subsystems and admin forced commands */
+	if (!s->is_subsystem && options.adm_forced_command == NULL &&
+	    (stat(_PATH_SSH_USER_RC, &st) >= 0)) {
 		snprintf(cmd, sizeof cmd, "%s -c '%s %s'",
 		    shell, _PATH_BSHELL, _PATH_SSH_USER_RC);
 		if (debug_flag)
